### PR TITLE
refactor: use turborepo

### PR DESCRIPTION
**Pull request by bkrem**
## Changes

- Configures turborepo
- Configures changeset
- Configures CI matrix to work atomically with `turbo`
- Removes dependency on `lerna`
- Fixes builds failing if package.json does not specify `dependencies`
- Unifies typescript version used across the monorepo pkgs
- Ensure that turborepo is aware of transitive dependencies on internal pkgs (e.g. `jsonrpc/provider -> @walletconnect/utils -> @walletconnect/window-getters`) by specifying them as explicit `devDependencies`

## TODO

- [x] Adjust the "Setup" section of the readme towards `turbo`
- [x] Adjust the "publishing packages" section of the readme for the new `changeset` flow

### CODE REVIEW

File: .changeset/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Changesets
```
Comment (ganchoradkov) [inline .changeset/README.md:1]: are to start including a changeset for releases or?

File: package.json [hunk 2]
```diff
@@ -12,24 +12,28 @@
...
+  "workspaces": [
+    "browser/*",
+    "jsonrpc/*",
+    "crypto/*",
+    "misc/*",
+    "relay/*"
+  ],
   "scripts": {
-    "clean": "lerna clean --yes",
-    "bootstrap": "lerna bootstrap",
```
Comment (ganchoradkov) [inline package.json:17]: >  '-   "bootstrap": "lerna bootstrap",'
I think we should remove the step from the main `README`

Comment (bkrem) [reply package.json:17]: Good callout, adding to my docs rewrite todos above 👍

File: .changeset/README.md (hunk 1 above)
Comment (bkrem) [reply .changeset/README.md:1]: Seems like the convention is to keep this file after looking at other big name projects that use it, e.g. https://github.com/emotion-js/emotion/tree/main/.changeset

**Review by devceline — COMMENTED** [review]
- Deleted my existing `walletconnect-utils` repo
- Recloned with this branch
- `npm i` 
- I get changes to my root `package-lock` 

The changes are adding packages (from this monorepo) that are dependencies from other packages in this monorepo into their `devDependencies` in the package-lock. Not sure if I messed anything up there

`npm -v === 9.19.2`
`node -v === 18.12.1`

**Comment by bkrem** [comment]
> * Deleted my existing `walletconnect-utils` repo
> * Recloned with this branch
> * `npm i`
> * I get changes to my root `package-lock`
> 
> The changes are adding packages (from this monorepo) that are dependencies from other packages in this monorepo into their `devDependencies` in the package-lock. Not sure if I messed anything up there
> 
> `npm -v === 9.19.2` `node -v === 18.12.1`

Thanks, good catch 👍 Should be fixed. 

I hadn't rerun `npm i` yet on my end after fixing some issues around transitive dependencies, i.e.:

> Ensure that turborepo is aware of transitive dependencies on internal pkgs (e.g. jsonrpc/provider -> @walletconnect/utils -> @walletconnect/window-getters) by specifying them as explicit devDependencies

File: package.json [hunk 3]
```diff
@@ -12,24 +12,28 @@
...
-    "test": "lerna run test",
-    "check": "run-s prettier lint build test",
+    "lint": "turbo lint",
+    "prettier": "turbo prettier",
+    "format": "turbo format",
+    "build": "turbo build",
+    "test": "turbo test",
+    "check": "turbo run prettier lint build test",
+    "check:force": "turbo run prettier lint build test --force=true",
     "reset": "run-s clean bootstrap check",
```
Comment (ganchoradkov) [inline package.json:30]:
```suggestion
```
I think this one is a leftover command from lerna as well

**Review by ganchoradkov — APPROVED** [review]
Awesome 💯 it works so smooth, and the cached runs are 🏎️🏎️🏎️🏎️

File: package.json (hunk 3 above)
Comment (bkrem) [reply package.json:30]: Yup good catch thank you!

**Pull request merged**